libs/hdf-converters/src/msft-secure-score-mapper.ts

```diff
@@ -0,0 +1,203 @@
+import {
+  SecureScore,
+  ControlScore,
+  SecureScoreControlProfile
+} from '@microsoft/microsoft-graph-types';
+import {ExecJSON} from 'inspecjs';
+import {version as HeimdallToolsVersion} from '../package.json';
+import {BaseConverter, ILookupPath, MappedTransform} from './base-converter';
+
+export class MsftSecureScoreMapper extends BaseConverter {
+  withRaw: boolean;
+  profiles: SecureScoreControlProfile[];
+
+  private getProfiles(controlName: string): SecureScoreControlProfile[] {
+    return this.profiles.filter((profile) => profile.id === controlName);
+  }
+
+  mappings: MappedTransform<
+    ExecJSON.Execution & {passthrough: unknown},
+    ILookupPath
+  > = {
+    platform: {
+      name: 'Heimdall Tools',
+      release: HeimdallToolsVersion
+    },
+    version: HeimdallToolsVersion,
+    statistics: {},
+    profiles: [
+      {
+        name: 'Microsoft Secure Score Scan',
+        title: {
+          transformer: (data: SecureScore) =>
+            `Azure Secure Score report - TenantID: ${data.azureTenantId}`
+        },
+        supports: [],
+        attributes: [],
+        groups: [],
+        status: 'loaded',
+        controls: [
+          {
+            path: 'controlScores',
+            id: {
+              transformer: (data: ControlScore) =>
+                `${data.controlCategory}:${data.controlName}`
+            },
+            title: {
+              transformer: (data: ControlScore) => {
+                const titles = this.getProfiles(data.controlName || '')
+                  .filter((profile) => profile.title !== undefined)
+                  .map((profile) => profile.title);
+
+                if (titles.length > 0) {
+                  return titles.join('... ');
+                } else {
+                  return `${data.controlCategory || ''}:${data.controlName || ''}`;
+                }
+              }
+            },
+            desc: {path: 'description'},
+            impact: {
```

Comment:
the types also show a userImpact field which is something that @ejaronne mentioned in his review.  there isn't a comment associated with it in the types, but it seems to have been provided in many of the profiles.

Reply:
`User impact is what the users will experience if the recommended action is enacted, and the Users affected are the people who will be impacted.`

https://learn.microsoft.com/en-us/defender-xdr/microsoft-secure-score-improvement-actions#assess-information-and-review-user-impact

Reply:
Ok so irrelevant info for us, but still something that should be put in the tags.